pynamodb/connection/base.py

```diff
@@ -95,6 +95,23 @@ def hash_keyname(self):
                     break
         return self._hash_keyname
 
+    def get_key_names(self, index_name=None):
+        """
+        Returns the names of the primary key attributes and index key attributes (if index_name is specified)
+        """
+        key_names = [self.hash_keyname]
+        range_keyname = self.range_keyname
+        if range_keyname is not None:
+            key_names.append(range_keyname)
+        if index_name is not None:
```

Comment:
is there not more than one index?

Reply:
you need to get the keys on a per-index basis since they are different for each index

Reply:
i think that's what i'm asking... shouldn't there be a loop here?

Reply:
the function (as written) returns keys for a specific index (or for the table if no index is specified), not all combinations for all indexes